File: 11containerWithMostWater/containerWithMostWater.java
class Solution {
    public int maxArea(int[] height) {
        var left = 0;
        var right = height.length - 1;
        var result = 0;
        while (left != right)
        {
            result = Math.max(result, (right - left) * IntStream.of(left, right).map($ -> height[$]).min().orElse(0));
            var $$ = IntStream.of(left, right).map($ -> height[$]).reduce((a,b) -> a - b).orElse(0) < 0 ? ++left : --right;
        }
        return result;
    }
}

File: 11containerWithMostWater/containerWithMostWater.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let [left, right] = [0, height.length - 1]
    let result = 0
    while (!Object.is(left, right))
    {
        result = Math.max(result, (right - left) * Math.min(...[left, right].map(_ => height[_])));
        [left, right].map(_ => height[_]).reduce((a,b) => a - b) < 0 ? ++left : --right
    }
    return result
};
